assets/components/productPage/productPagePeriodForm/productPagePeriodFormReducer.js

```diff
@@ -0,0 +1,26 @@
+// @flow
+// ----- Imports ----- //
+import type { Action } from './productPagePeriodFormActions';
+
+export type State<Period> = {
+  period: Period | null;
+};
+
+// ----- Reducer ----- //
+function productPagePeriodFormReducerFor<Period>(scope: string, initialState: State<Period>) {
```

Comment:
I think you could also pass in `period` as the second argument here, and then define the initial state in this function:

```jsx
function productPagePeriodFormReducerFor<Period>(scope: string, period: Period) {

  const initialState = { period };
  ...
}
```

That way the initial state is defined alongside the reducer and its `State` type. I don't know if this is better, it's just an alternative.

Reply:
your suggestion makes the whole module a bit more portable considering the state object is always encapsulated outside of this, I like it! 👍